doc/string/slices.rdoc

```diff
@@ -0,0 +1,144 @@
+== \String Slices
+
+A _slice_ of a string is a substring that is selected by certain criteria.
+
+These instance methods make use of slicing:
+
+- String#[] (and its alias String#slice) return a slice copied from +self+.
+- String#[]= returns a copy of +self+ with a slice replaced.
+- String#slice! returns +self+ with a slice replaced.
+
+Each of the above methods takes arguments that determine the slice
+to be copied or replaced.
+
+The arguments have several forms.
+For string +string+,  the forms are:
+
+- <tt>string[index]</tt>.
+- <tt>string[start, length]</tt>.
+- <tt>string[range]</tt>.
+- <tt>string[regexp, capture = 0]</tt>.
+- <tt>string[substring]</tt>.
+
+<b><tt>string[index]</tt></b>
+
+When non-negative integer argument +index+ is given,
+the slice is the 1-character substring found in +self+ at character offset +index+:
+
+  'bar'[0]       # => "b"
+  'bar'[2]       # => "r"
+  'bar'[20]      # => nil
+  'тест'[2]      # => "с"
+  'こんにちは'[4]  # => "は"
+
+When negative integer +index+ is given,
+the slice begins at the offset given by counting backward from the end of +self+:
+
+  'bar'[-3]         # => "b"
+  'bar'[-1]         # => "r"
+  'bar'[-20]        # => nil
+
+<b><tt>string[start, length]</tt></b>
+
+When non-negative integer arguments +start+ and +length+ are given,
+the slice begins at character offset +start+, if it exists,
+and continues for +length+ characters, if available:
+
+  'foo'[0, 2]       # => "fo"
+  'тест'[1, 2]      # => "ес"
+  'こんにちは'[2, 2]  # => "にち"
+  # Zero length.
+  'foo'[2, 0]       # => ""
+  # Length not entirely available.
+  'foo'[1, 200]     # => "oo"
+  # Start out of range.
+  'foo'[4, 2]      # => nil
+  'foo'[-4, 2]     # => nil
```

Comment:
I think this example should go below to the "When negative +start+ and non-negative +length+" section.

Reply:
Done.